pkg/controller/keycloak/keycloak_reconciler.go

```diff
@@ -66,6 +66,12 @@ func (i *KeycloakReconciler) reconcileExternalDatabase(desired *common.DesiredCl
 
 func (i *KeycloakReconciler) reconcileExternalAccess(desired *common.DesiredClusterState, clusterState *common.ClusterState, cr *kc.Keycloak) {
 	if !cr.Spec.ExternalAccess.Enabled {
+		if clusterState.KeycloakRoute != nil {
```

Comment:
I totally agree, we should have something like this in the Operator. However, I think the implementation should be different.

My preference would be to inspect objects with owning reference pointing to the Operator (all objects that were created by this Operator has owning reference pointing to it). If any of those objects is not present in the Secondary Resources (from the status field of the CR), they should be removed. 

This way we would have a generic mechanism for removing objects that are no longer necessary. 

@pb82 @davidffrench @mhajas WDYT?

Reply:
That would be good to have @slaskawi . I don't believe it will fix this issue as the route would exist in the secondary resources in the status block. The generic solution would work for outside sources adding the operator as an owner reference. Although I don't know if i have a problem with that either.

This code will fix an issue where you are switching the CR.spec.externalaccess.enabled from true to false.